src/firmware/app/world/firmware_robot.c

```diff
@@ -229,7 +241,7 @@ void velocity_wheels_setLocalVelocity(
     float robot_velocity[3];
     robot_velocity[0] = linear_velocity_x;
     robot_velocity[1] = linear_velocity_y;
-    robot_velocity[2] = angular_velocity;
+    robot_velocity[2] = angular_velocity * ROBOT_RADIUS;
     float wheel_velocity[4];
     speed3_to_speed4(robot_velocity, wheel_velocity);
```

Comment:
I thought speed3_to_speed4 took angular velocity?

Reply:
I'm guessing a bit here at the proper usage of this, but I followed these examples:
[Example 1](https://github.com/UBC-Thunderbots/Software/blob/08cf8f695c57af10871b66d0d86e0cb483eda6b5/src/firmware/app/control/control.c#L81)
[Example 2](https://github.com/UBC-Thunderbots/Software/blob/08cf8f695c57af10871b66d0d86e0cb483eda6b5/src/firmware/app/control/control.c#L149)

Both of these instances, although dealing with force instead of speed, convert them to be the same unit (N) before applying the matrix. This seems to be in line with documentation, which mentions unitless conversion ([here](https://github.com/UBC-Thunderbots/Software/blob/08cf8f695c57af10871b66d0d86e0cb483eda6b5/src/firmware/shared/physics.c#L38)).

Also side note - I noticed that `speed3_to_speed4` actually uses the force matrix and `force3_to_force4` uses force matrix ([here](https://github.com/UBC-Thunderbots/Software/blob/08cf8f695c57af10871b66d0d86e0cb483eda6b5/src/firmware/shared/physics.c#L56))

Reply:
well force3_to_force4 uses the speed matrix... I'm not sure what's right exactly, so given it works in sim maybe it's ok for now

Reply:
oops yeah meant force3_to_force4 uses speed matrix, I don't quite think that's correct, but who knows